local/ntap/workflows-560-rnaseq-human.py

```diff
@@ -0,0 +1,213 @@
+    output_number: int
+    """The number for the output, in some cases this is the same as the run_number, 
+    but in other cases it isn't because the pipeline has been re-ran."""
+
+    bucket_name: str
+    """The name of the bucket to stage the samplesheet in."""
+
+    staging_key: str
+    """The key in the S3 bucket where this workflow is going to run."""
+
+    @property
+    def samplesheet_location(self) -> str:
+        """The location where the unstaged samplesheet is located."""
+        return f"{self.samplesheet_location_prefix}{self.samplesheet}"
+
+    @property
+    def samplesheet_to_stage_key(self) -> str:
+        """The key in the S3 bucket where the samplesheet is going to be staged."""
+        return f"{self.staging_key}to_stage/{self.samplesheet}"
+
+    @property
+    def staged_samplesheet_location(self) -> str:
+        """The S3 uri where the samplesheet is staged."""
+        return f"{self.staging_location}synstage_{self.id}/{self.samplesheet}"
+
+    @property
+    def staging_location(self) -> str:
+        """The S3 uri where the workflow is going to be run."""
+        return f"s3://{self.bucket_name}/{self.staging_key}"
+
+    @property
+    def samplesheet_location_prefix(self) -> str:
+        """The S3 uri where the unstaged samplesheet is located."""
+        return f"s3://{self.bucket_name}/{self.staging_key}to_stage/"
+
+    @property
+    def output_directory(self) -> str:
+        """The S3 uri where the output is going to be uploaded to. The is used as the
+        input for the synindex workflow."""
+        return f"s3://{self.bucket_name}/outputs/rnaseq_GRCh38_{self.id}_{self.output_number}/"
+
+    @property
+    def synstage_run_name(self) -> str:
+        """The name of the synstage run."""
+        return f"synstage_{self.id}"
+
+    @property
+    def rnaseq_run_name(self) -> str:
+        """The name of the rnaseq run."""
+        return f"rnaseq_GRCh38_{self.id}_{self.run_number}"
+
+    @property
+    def synindex_run_name(self) -> str:
+        """The name of the synindex run."""
+        return f"synindex_{self.id}_{self.run_number}"
+
+
+def generate_datasets() -> list[Dataset]:
+    """Generate list of datasets.
+
+    Source: https://sagebionetworks.jira.com/browse/WORKFLOWS-538
+
+    This is expecting that the SampleSheet to be run exists in the `prefix` location.
+    """
+    return [
+        Dataset(
+            id="syn53972620",
+            samplesheet="syn51198895_RNA_Seq_SampleSheet.csv",
+            staging_key="samplesheets/RNASeq_Process/WORKFLOWS-560/",
+            bucket_name="ntap-add5-project-tower-bucket",
+            synapse_id_for_output="syn55228511",
+            run_number=1,
+            output_number=1,
+        )
+    ]
+
+def stage_samplesheet(syn: Synapse, dataset: Dataset):
+    """Download the samplesheet from synapse and upload it to S3 in the location where synstage
+    is going to grab the file.
+
+    :param syn: The logged in synapse instance.
```

Comment:
Return type of None. And could you change these over to google style docstrings. That'll kick off the overall conversion to it for this repo.

Reply:
Thanks for catching it. I updated it in the latest commit.